File: library/src/main/java/com/mikepenz/materialdrawer/model/interfaces/Checkable.java
package com.mikepenz.materialdrawer.model.interfaces;

/**
 * Created by mikepenz on 03.02.15.
 */
public interface Checkable<T> {
    public T withCheckable(boolean checkable);

    public boolean isCheckable();

    public void setCheckable(boolean checkable);
}

File: materialdrawer/src/main/java/com/mikepenz/materialdrawer/model/interfaces/Checkable.kt
package com.mikepenz.materialdrawer.model.interfaces

/**
 * Defines a [IDrawerItem] with support for being selected
 */
interface Checkable : Selectable {
    var isChecked: Boolean
}

@Deprecated("Please consider to replace with the actual property setter")
fun <T : Checkable> T.withChecked(checked: Boolean): T {
    this.isChecked = checked
    return this
}


@Deprecated("Please consider to replace with the actual property setter")
fun <T : Checkable> T.withCheckable(checkable: Boolean): T {
    this.isSelectable = checkable
    return this
}
